packages/dosespot-react/src/useDoseSpotIFrame.ts

```diff
@@ -0,0 +1,70 @@
+import { Patient } from '@medplum/fhirtypes';
+import { useMedplum, useResource } from '@medplum/react-hooks';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { DOSESPOT_IFRAME_BOT, DOSESPOT_PATIENT_ID_SYSTEM, DOSESPOT_PATIENT_SYNC_BOT } from './common';
+
+export interface DoseSpotIFrameOptions {
+  readonly patientId?: string;
+  readonly onPatientSyncSuccess?: () => void;
+  readonly onPatientSyncError?: (err: unknown) => void;
+  readonly onIframeSuccess?: (url: string) => void;
+  readonly onIframeError?: (err: unknown) => void;
+}
+
+export function useDoseSpotIFrame(options: DoseSpotIFrameOptions): string | undefined {
+  const medplum = useMedplum();
+  const { patientId, onPatientSyncSuccess, onPatientSyncError, onIframeSuccess, onIframeError } = options;
+  const patient = useResource<Patient>(patientId ? { reference: `Patient/${patientId}` } : undefined);
+  const initializingRef = useRef<boolean>(false);
+  const [iframeUrl, setIframeUrl] = useState<string | undefined>(undefined);
+
+  const onPatientSyncSuccessRef = useRef(onPatientSyncSuccess);
```

Comment:
I like this pattern a lot 👍

Reply:
🎉  I definitely think of this as a Derrick-ism 😎